src/VisionModel/resnet.jl

```diff
@@ -0,0 +1,85 @@
+        Conv((3, 3), outchannels[1] => outchannels[2], stride = 1, pad = 1),
+        BatchNorm(outchannels[2], λ=relu)) : 
+  Chain(Conv((3, 3), inplanes => outchannels[1], stride = 1),
+        BatchNorm(outchannels[1], λ = relu),
+        Conv((3, 3), outchannels[1] => outchannels[2], stride = 1, pad = 1),
+        BatchNorm(outchannels[2], λ=relu))
+
+bottleneck(inplanes::Int, outchannels, downsample::Bool = false) = downsample ?
+  Chain(Conv((1, 1), inplanes => outchannels[1], stride = 2),
+        BatchNorm(outchannels[1], λ = relu),
+        Conv((3, 3), outchannels[1] => outchannels[2], stride = 1, pad = 1),
+        BatchNorm(outchannels[2], λ = relu),
+        Conv((1, 1), outchannels[2] => outchannels[3], stride = 1),
+        BatchNorm(outchannels[3], λ = relu)) :
+  Chain(Conv((1, 1), inplanes => outchannels[1], stride = 1),
+        BatchNorm(outchannels[1], λ = relu),
+        Conv((3, 3), outchannels[1] => outchannels[2], stride = 1, pad = 1),
+        BatchNorm(outchannels[2], λ = relu),
+        Conv((1, 1), outchannels[2] => outchannels[3], stride = 1),
+        BatchNorm(outchannels[3], λ = relu))
+
+projection(inplanes::Int, outplanes::Int, stride::Int) = Chain(Conv((1, 1), inplanes => outplanes, stride=stride),
+                                                               BatchNorm((outplanes), λ=relu))
+
+identity(inplanes::Int, outplanes::Int, stride::Int) = inplanes < outplanes ? 
+  Array -> PaddedView(0, Array, outplanes) : +
+
+function resnet(block, shortcut_config, channel_config, block_config)
+  layers = []
+  push!(layers, Conv((7, 7), 3=>inplanes, stride=(2, 2), pad=(3, 3)))
+  push!(layers, BatchNorm(inplanes, λ=relu))
+  push!(layers, MaxPool((3, 3), stride=(2, 2), pad=(1, 1)))
+  inplanes = 64
+  baseplanes = 64
+  for nrepeats in block_config
+    outplanes = baseplanes .* channel_config
+    push!(layers, block(inplanes, outplanes, true))
+    if shortcut_config == :A
+      push!(layers, SkipConnection(block(inplanes, outplanes, true), 
+                                   identity(inplanes, outplanes, 2)))
+    elseif shortcut_config == :B || shortcut_config == :C
+      push!(layers, SkipConnection(block(inplanes, outplanes, true),
+                                   projection(inplanes, outplanes[end], 2)))
+    end
+    inplanes = outplanes[end]
+    for i in 1:nrepeats
+      if shortcut_config == :A || shortcut_config == :B
+        push!(layers, SkipConnection(block(inplanes, outplanes, false),
+                                     identity(inplanes, outplanes[end], 2)))
+      elseif shortcut_config == :C
+        push!(layers, SkipConnection(block(inplanes, outplanes, false),
+                                     projection(inplanes, outplanes, 2)))
+      end
+      inplanes = outplanes[end]
+    end
+    baseplanes *= 2
+  end
+  push!(layers, AdaptiveMeanPool(1, 1))
+  push!(layers, x -> flatten(x, 1))
+  push!(layers, Dense(512 * expansion, 1000))
```

Comment:
Instead of `expansion` you can use `inplanes` here (which should equal the value of last repetition's final conv output channels.